mlflow/evaluation/utils.py

```diff
@@ -0,0 +1,191 @@
+    }
+
+
+def _get_empty_evaluations_dataframe() -> pd.DataFrame:
+    """
+    Creates an empty DataFrame with columns for evaluation data.
+    """
+    schema = _get_evaluations_dataframe_schema()
+    df = pd.DataFrame(columns=schema.keys())
+    return _apply_schema_to_dataframe(df, schema)
+
+
+def _get_assessments_dataframe_schema() -> Dict[str, str]:
+    """
+    Returns the pandas schema for the assessments DataFrame.
+    """
+    return {
+        "evaluation_id": "string",
+        "name": "string",
+        "source": "object",
+        "timestamp": "int64",
+        "boolean_value": "object",
+        "numeric_value": "object",
+        "string_value": "object",
+        "rationale": "object",
+        "metadata": "object",
+        "error_code": "object",
+        "error_message": "object",
+    }
+
+
+def _get_empty_assessments_dataframe() -> pd.DataFrame:
+    """
+    Creates an empty DataFrame with columns for evaluation assessments data.
+    """
+    schema = _get_assessments_dataframe_schema()
+    df = pd.DataFrame(columns=schema.keys())
+    return _apply_schema_to_dataframe(df, schema)
+
+
+def _get_metrics_dataframe_schema() -> Dict[str, str]:
+    """
+    Returns the pandas schema for the metrics DataFrame.
+    """
+    return {
+        "evaluation_id": "string",
+        "key": "string",
+        "value": "float64",
+        "timestamp": "int64",
+    }
+
+
+def _get_empty_metrics_dataframe() -> pd.DataFrame:
+    """
+    Creates an empty DataFrame with columns for evaluation metric data.
+    """
+    schema = _get_metrics_dataframe_schema()
+    df = pd.DataFrame(columns=schema.keys())
+    return _apply_schema_to_dataframe(df, schema)
+
+
+def _get_tags_dataframe_schema() -> Dict[str, str]:
+    """
+    Returns the pandas schema for the tags DataFrame.
+    """
+    return {
+        "evaluation_id": "string",
+        "key": "string",
+        "value": "string",
+    }
+
+
+def _get_empty_tags_dataframe() -> pd.DataFrame:
+    """
+    Creates an empty DataFrame with columns for evaluation tags data.
+    """
+    schema = _get_tags_dataframe_schema()
+    df = pd.DataFrame(columns=schema.keys())
+    return _apply_schema_to_dataframe(df, schema)
+
+
+def _apply_schema_to_dataframe(df: pd.DataFrame, schema: Dict[str, str]) -> pd.DataFrame:
+    """
+    Applies a schema to a DataFrame.
+
+    Args:
+        df (pd.DataFrame): DataFrame to apply the schema to.
+        schema (Dict[str, Any]): Schema to apply.
+
+    Returns:
+        pd.DataFrame: DataFrame with schema applied.
+    """
+    for column in df.columns:
+        df[column] = df[column].astype(schema[column])
+    # By default, null values are represented as `pd.NA` in pandas when reading a dataframe from
+    # JSON. However, MLflow entities use `None` to represent null values. Accordingly, we convert
+    # instances of pd.NA to None so that DataFrame rows can be parsed as MLflow entities
+    return df.replace(pd.NA, None)
```

Comment:
Does the pandas `<NA>` type definition for missing data apply to the numpy `int64` type? I thought that that feature was only for declaring the pandas wrapper `"Int64"` (upper case "I") and that a missing value in a numpy-type would be represented by `NaN` and cast to a floating point number?
Or does pd.NA cover the numpy NaN as well? If yes, disregard this comment. 
https://pandas.pydata.org/docs/user_guide/integer_na.html

Reply:
Brilliant question!

`pd.NA` does not cover numpy NaN / Python NaN, but this is a good thing. NaN is an accepted / legitimate value for MLflow metrics.

I'll follow up with some additional test coverage for this edge case in a subsequent PR and merge for now. Thanks @BenWilson2 !